docs/DeveloperGuide.md

```diff
@@ -255,6 +255,162 @@ _{more aspects and alternatives to be added}_
+        * `RemoveAbortion#execute()` checks if the `no` input is valid, calling `RemoveConfirmation#isValidInput()`
+        * `RemoveConfirmation#isValidInput()` will return `true` if the input is valid, and `false` otherwise.
+            * Validity of input is determined by the previous command executed by the user - a valid `remove INDEX` 
+          command, that serves as a precursor to the removal abortion process.
+  * If the user aborts the removal with `no`, `RemoveCommand` will abort the removal process.
+      * The default list of contacts will be shown with the text input of the `CommandBox` cleared, and `RemoveAbortion` 
+    will provide feedback on the abortion of the removal process.
+
+
+#### Design considerations:
+
+Several design considerations were taken into account when implementing the safe-removal feature.
+
+_**FIRST CATEGORY**: For shortlisting the contact to be removed and seeking confirmation of the contact to be removed:_ 
+
+
+* **Alternative 1 (current choice)**: To use the same command word (i.e. `remove` - `remove NAME` and `remove INDEX`)
+to perform the shortlisting of contacts with matching names, as well as the confirmation of the contact to be removed.
+  * Pros: Simplifies the command structure, more intuitive for users to approach removal process
+    * Using the same command word `remove` for both shortlisting and confirmation processes reduces the cognitive load,
+    allowing the process to be more user-friendly
+  * Cons: May lead to ambiguity in the command execution process to new developers who are used to the conventions set
+  by other commands, as this command structure makes use of an overloaded `RemoveCommand` constructor
+
+
+* **Alternative 2**: To use the existing `find` command to shortlist the contact to be removed, then use the `delete`
+command to perform the deletion
+* Pros: Separates the shortlisting and confirmation processes
+  * This reduces ambiguity in the command execution process for future developers
+* Cons: Require 2 different commands for deletion 
+  * Increasing the number of commands required to perform the removal process, makes it less intuitive for users to
+      approach the removal process. Furthermore, when it comes to **removing contacts**, users might find it a lot more 
+        intuitive (as it is self-explanatory) to use a`delete`/`remove` command instead of having to use `find` first, 
+        where in a natural logical context, most users would only use `find` if they are simply looking for a contact.
+
+
+**Decision**: 
+Weighing the pros and cons of Alternatives 1 and 2, we have decided to go with **Alternative 1** due to the enhanced 
+user experience it provides, making the removal process more intuitive and user-friendly. 
+
+**Other considerations**:
+1. **Single Responsibility Principle**: The `RemoveCommand` class is designed to handle both the shortlisting and 
```

Comment:
Good justification, I do agree that SRP is not broken even with the current implementation!